src/fprime_gds/common/communication/ccsds/space_data_link.py

```diff
@@ -30,12 +32,31 @@ class SpaceDataLinkFramerDeframer(FramerDeframer):
     )
     CRC_CALCULATOR = crc.Calculator(CRC_CCITT_CONFIG)
 
+    # For backwards compatibility if not found in dictionary (loaded by ConfigManager)
+    FALLBACK_SCID = 0x44
+    FALLBACK_FRAME_SIZE = 1024
+
+
     def __init__(self, scid, vcid, frame_size):
-        """ """
-        self.scid = scid
-        self.vcid = vcid
-        self.frame_size = frame_size
+        """Initialize with the given spacecraft id, virtual channel id, and frame size.
+        If scid or frame_size are None, they will be pulled from ConfigManager constants
+        if present, or use fallback values."""
+        dict_scid = None
+        dict_frame_size = None
+        try:
+            dict_scid = ConfigManager.get_instance().get_constant("ComCfg.SpacecraftId")
+        except ConfigBadTypeException:
+            pass # Config value not found, move on
+        try:
+            dict_frame_size = ConfigManager.get_instance().get_constant("ComCfg.TmFrameFixedSize")
+        except ConfigBadTypeException:
+            pass # Config value not found, move on
+
         self.sequence_number = 0
+        self.vcid = vcid
```

Comment:
Why is vcid not also in dictionary configuration?

Reply:
The reason we have Spacecraft ID in `ComCfg` but not VCID is because Spacecraft ID is tied to a FSW build, where VCID is more loosely defined by the standard (on purpose). One GDS<=>SC link can communicate over multiple Virtual channels, we just happen to only support one channel in F´ currently. 
A project might choose to implement more complex framers than we do, in which case a single `VcIdValue` in the model doesn't make sense anymore.
The standard does say "TM frame size is fixed for the lifespan of the mission" hence putting it as a single constant in the model.